File: springwolf-examples/springwolf-amqp-example/src/test/java/io/github/stavshamir/springwolf/example/ApiIntegrationTests.java
package io.github.stavshamir.springwolf.example;

import org.json.JSONException;
import org.junit.jupiter.api.Test;
import org.skyscreamer.jsonassert.JSONAssert;
import org.skyscreamer.jsonassert.JSONCompareMode;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.boot.test.web.client.TestRestTemplate;
import org.testcontainers.shaded.org.apache.commons.io.IOUtils;

import java.io.IOException;
import java.io.InputStream;
import java.nio.charset.StandardCharsets;

@SpringBootTest(classes = {SpringwolfExampleApplication.class}, webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
public class ApiIntegrationTests {

    @Autowired
    private TestRestTemplate restTemplate;

    @Value("${server.port}")
    public Integer serverPort;

    @Test
    public void asyncApiResourceArtifactTest() throws JSONException, IOException {
        String url = "/springwolf/docs";
        String actual = restTemplate.getForObject(url, String.class);
        System.out.println("Got: " + actual);

        InputStream s = this.getClass().getResourceAsStream("/asyncapi.json");
        String expected = IOUtils.toString(s, StandardCharsets.UTF_8);

        JSONAssert.assertEquals(expected, actual, JSONCompareMode.STRICT_ORDER);
    }
}

File: springwolf-examples/springwolf-amqp-example/src/test/java/ApiIntegrationTests.kt
import org.apache.commons.io.IOUtils
import org.junit.ClassRule
import org.junit.Test
import org.skyscreamer.jsonassert.JSONAssert
import org.skyscreamer.jsonassert.JSONCompareMode
import org.springframework.web.client.RestTemplate
import org.testcontainers.containers.DockerComposeContainer
import java.io.File
import java.io.InputStream
import java.nio.charset.StandardCharsets


class ApiIntegrationTests {

    companion object {
        private val restTemplate = RestTemplate()

        private const val APP_NAME = "app_1"
        private const val APP_PORT = 8080

        @JvmField
        @ClassRule
        val environment: DockerComposeContainer<Nothing> = DockerComposeContainer<Nothing>(File("docker-compose.yml"))
                .withExposedService(APP_NAME, APP_PORT)
    }

    private fun baseUrl(): String {
        val host = environment.getServiceHost(APP_NAME, APP_PORT)
        val port = environment.getServicePort(APP_NAME, APP_PORT)
        return "http://$host:$port"
    }

    @Test
    fun `asyncapi-docs shold return the correct json response`() {
        val url = "${baseUrl()}/springwolf/docs"
        val actual = restTemplate.getForObject(url, String::class.java)

        val s: InputStream = this.javaClass.getResourceAsStream("asyncapi.json")
        val expected: String = IOUtils.toString(s, StandardCharsets.UTF_8)

        JSONAssert.assertEquals(expected, actual, JSONCompareMode.NON_EXTENSIBLE)
    }

}